symphony-bdk-template/symphony-bdk-template-api/src/main/java/com/symphony/bdk/template/api/Template.java

```diff
@@ -0,0 +1,18 @@
+package com.symphony.bdk.template.api;
+
+import java.util.Map;
+
+/**
+ * Interface to represent a template.
+ * A template takes parameters in input and outputs a string.
+ */
+public interface Template {
+
+  /**
+   *
+   * @param parameters the map which contains the parameter names as map key and the parameter values as map values
+   * @return the generated string where each parameter is replaced by its value provided in the map
+   * @throws TemplateException in case of issues during the string generation, e.g. missing parameter
+   */
+  String process(Map<String, String> parameters) throws TemplateException;
```

Comment:
With a `Map<String, String>` as parameter we can only manage 1 level of attributes when most of the template engines support tree values as data-model (cf. https://freemarker.apache.org/docs/dgui_quickstart_datamodel.html).

Reply:
Freemarker can process an object for instance (https://freemarker.apache.org/docs/api/freemarker/template/Template.html#process-java.lang.Object-java.io.Writer-), that would be nice to study different template engines to see what they offer but I think this is where we should go

Reply:
👍

Reply:
FreeMarker accepts any kind of objects, like Map<String, Object> or java beans: https://freemarker.apache.org/docs/pgui_quickstart_createdatamodel.html
Velocity accepts a Context object which is a Map like object, see : http://velocity.apache.org/engine/devel/developer-guide.html#the-context
I would suggest to accept a Object in the interface and each implementation will have to check the actual type.